### PR TITLE
feat(operator): add last operator and update marble tests to support grouping

### LINKED ISSUE #304: Marble diagram tests just don't work for synchronous scenarios
For example: `Observable.of(1,2,3)` ... how are you going to represent 3 values and a completion in 1 frame? `123|`? and then how does that "line up" to expected results? What about 3 values and an error? what about 100ms, then 14 synchronous values, then 200ms then one value and it completes?

It's a shame because I find the look of the test DSL very compelling.

@mattpodwysocki can say "I told you so". :p

cc/@staltz and @trxcllnt

### LINKED ISSUE #306: Add last operator
Just what it says, we need the `last` operator.

### PR DESCRIPTION
**Pull request by blesh**
- adds EmptyError error type
- adds ability to group values in marble tests
- adds last operator

closes #304
closes #306

### CODE REVIEW

**Comment by blesh** [comment]
now the marble tests support grouping.. where the `(` designates the frame a group is emitted on, and it gathers all of the values until it finds a `)` ... nested groups are *not* supported.

```
merging this:

----a------|
----(bc)---|

makes this:

----(abc)--|
```

**Comment by blesh** [comment]
attn @kwonoj @staltz

File: spec/operators/last-spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* globals describe, it, expect, expectObservable, hot, cold */
+var Rx = require('../../dist/cjs/Rx');
+
+describe('Observable.prototype.last()', function(){
+  it('should take the last value of an observable', function(){
+    var e1 = hot('--a--^--b--c--|');
+    var expected =    '---------(c|)';
```
Comment (kwonoj) [inline spec/operators/last-spec.js]: Recommend to unclose parenthesis when completed?

Comment (blesh) [inline spec/operators/last-spec.js]: That's a nexted value and a completion in the same frame

**Pull request merged**